graf2d/win32gdk/src/TGWin32.cxx

```diff
@@ -4802,6 +4801,11 @@ Window_t TGWin32::CreateWindow(Window_t parent, Int_t x, Int_t y,
    GdkColor background_color;
    ULong_t xmask = 0;
 
+   /* copy these bits from TGFrame.h, otherwise one should link against gui lib */
+   UInt_t kMainFrame       = BIT(0);
```

Comment:
Don’t forget to move those in a common place.

Reply:
Is it necessary? Probably we should ask @bellenot - that is better solution here.

Reply:
What about TVirtualX? (as a quick idea...)

Reply:
@bellenot It is your decision. I could copy related  enum from TGFrame.h to core/gui/inc/GuiTypes.h - there are many other similar enums.

Reply:
Fine with me

Reply:
@linev BTW, just wondering what does that implies for users' code? Will they have to change their includes?

Reply:
@bellenot It is not a user problem - it is missing libs dependency in ROOT itself. Via this enum `win32gdk` have to have `Gui` in list of dependencies while it uses includes from `Gui`. It comes out when I tried to compile ROOT using only source dirs in include paths

Reply:
@linev I know that, and I agree, but I was just wondering if moving those declarations around could break users code at some point...

Reply:
Only if somebody, who do not include `TGFrame.h` before, uses `EFrameType` as enum or any constants from it. Therefore my effort with `less_includes` PR should pay here - normally `GuiTypes.h` should not appear in non-GUI classes.

Reply:
Here is my commit: https://github.com/root-project/root/pull/5231/commits/a9688b568eb358184ddbdaacf7b9a852a14d19d4

Reply:
Very good! Thanks @linev !